### PR TITLE
LPD-37264 Apply Root Model to the Data Set Object definition through the batch engine 

### PR DESCRIPTION
**Pull request by dsanz**
In this PR I'm applying all we discussed in the previous https://github.com/liferay-frontend/liferay-portal/pull/4511, namely:
* A new attribute `target` in Data Set Action, to store what we previously called `type`
* Now, `type` stores the action type (creation/item) thus making two relationships unnecessary
* There is a single relationship between Action and Data Set
* The above affects data creation, fecthing (DSM) and rendering (fragment)
* And of course, tests
* UI still uses "type" to refer to "target" which means that part remains

This is a breaking change in the model

### CODE REVIEW

**Comment by dsanz** [comment]
ci:test:playwright

File: modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts [hunk 1]
```diff
@@ -54,14 +55,14 @@ const skusDataSetConfig = {
 
 const tableSectionsDataSetConfig = {
 	name: getRandomString(),
-	restApplication: '/data-set-admin/table-sections',
+	restApplication: `${API_ENDPOINT_PATH}table-sections`,
```
Comment (juanjofgliferay) [inline modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts:58]: Missing `/`

Comment (dsanz) [reply modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dataSets.spec.ts:58]: This is the only pw tests in DSM that is failing, nice catch!

**Comment by dsanz** [comment]
just pushed the change @juanjofgliferay , re-running

**Comment by dsanz** [comment]
ci:test:playwright

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Review by markocikos — APPROVED** [review]
@dsanz LGTM! :sparkles:

**Comment by dsanz** [comment]
ci:reevaluate:1377401_18553

**Comment by dsanz** [comment]
I will rebase this one as I don't think ci results make any sense

**Comment by dsanz** [comment]
ci:forward

**Pull request closed without merging**